engine-standalone-storage/src/sync/mod.rs

```diff
@@ -2,21 +2,198 @@ use aurora_engine::parameters::SubmitArgs;
 use aurora_engine::pausables::{
     EnginePrecompilesPauser, PausedPrecompilesManager, PrecompileFlags,
 };
-use aurora_engine::{connector, engine, parameters::SubmitResult, state, xcc};
+use aurora_engine::{
+    connector, engine,
+    parameters::{self, SubmitResult},
+    state, xcc,
+};
 use aurora_engine_modexp::ModExpAlgorithm;
 use aurora_engine_sdk::env::{self, Env, DEFAULT_PREPAID_GAS};
+use aurora_engine_transactions::EthTransactionKind;
 use aurora_engine_types::{
     account_id::AccountId,
+    borsh::BorshDeserialize,
     parameters::PromiseWithCallbackArgs,
     types::{Address, Yocto},
     H256,
 };
+use std::str::FromStr;
 
 pub mod types;
 
 use crate::engine_state::EngineStateAccess;
 use crate::{BlockMetadata, Diff, Storage};
-use types::{Message, TransactionKind, TransactionMessage};
+use types::{Message, TransactionKind, TransactionKindTag, TransactionMessage};
+
+/// Try to parse an Aurora transaction from raw information available in a Near action
+/// (method name, input bytes, data returned from promises).
+#[allow(clippy::too_many_lines)]
+#[must_use]
+pub fn parse_transaction_kind(
+    method_name: &str,
+    bytes: Vec<u8>,
+    promise_data: &[Option<Vec<u8>>],
+) -> Option<TransactionKind> {
```

Comment:
Does it make sense to use a `Result` instead? It could help us to find out where we have a problem with deserialization. Because in the current version, if something goes wrong, we just get panic with the message: `All method names must be known by standalone` without locating a place where the problem exactly happens.

Reply:
I used `Option` because that is how it was in `borealis-engine-lib`, but I agree that it is nicer to return a `Result` instead. Done in https://github.com/aurora-is-near/aurora-engine/pull/810/commits/8b41e8e9eb70aeec32f35bfd8b98d1dc8ac73229